src/main/shell-session/local-shell-session/local-shell-session.ts

```diff
@@ -20,7 +29,14 @@ export class LocalShellSession extends ShellSession {
   }
 
   public async open() {
-    const env = await this.getCachedShellEnv();
+    let env = await this.getCachedShellEnv();
+  
+    if (this.entity) {
+      const ctx = { catalogEntity: this.entity };
+
+      env = JSON.parse(JSON.stringify(this.shellEnvModifiers.reduce((env, modifier) => modifier(ctx, env), env)));
```

Comment:
I meant this actually:

```suggestion
      env = this.shellEnvModifiers.reduce((env, modifier) => modifier(ctx, env), JSON.parse(JSON.stringify(env)));
```

Reply:
right!